### DISCUSSION TITLE
MaximumSize based eviction functionality is slow starting with version 3.1.2

### DISCUSSION THREAD

**Discussion by alexursea** [ben-manes/caffeine#1047] [Q&A] [answered]
Hi, 
After migrating from 2.8.5 to 3.1.6 (narrowed it down and it seems between 3.1.1 and 3.1.2 the problem appeared),
a JUNIT test failed that was checking, the first entry is evicted when the maximum size is exceeded by one entry.
Usually, it took maximum 100ms for the eviction to be successful, but now it is around 10 seconds.

What can I do in this case?
Thank you.
```
         Cache<CacheKey, MyEntry> cache = Caffeine.newBuilder()
                .maximumSize(2)
                .expireAfter(new CustomExpiry(100)
                .removalListener((key, value, cause) ->
                        log.info("Removing cached entry for key={}, removed value={}, cause={}.",
                                key, value, cause))
                .build();
```

**Comment by ben-manes** [comment, marked as answer]
I’d need to see a minimal test but it’s not wrong as described since we don’t use a policy like lru. For example if the workload is a loop then MRU is the optimal policy and the first entry would not be evicted. If random then all entries are equally good/bad to evict, so likewise it might keep the oldest. A pattern of popular entries would give the policy a reason to have a preference and evict the less valuable entries.

**Reply by alexursea** [reply]
Thank you, I thought you use LRU policy.
